packages/roosterjs-editor-plugins/lib/plugins/Paste/sourceValidations/getPasteSource.ts

```diff
@@ -0,0 +1,47 @@
+import documentContainWacElements from './documentContainWacElements';
+import isExcelDesktopDocument from './isExcelDesktopDocument';
+import isGoogleSheetDocument from './isGoogleSheetDocument';
+import isPowerPointDesktopDocument from './isPowerPointDesktopDocument';
+import isWordDesktopDocument from './isWordDesktopDocument';
+import shouldConvertToSingleImage from './shouldConvertToSingleImage';
+import { BeforePasteEvent, ClipboardData, IEditor } from 'roosterjs-editor-types';
+import { KnownSourceType } from './KnownSourceType';
+
+/**
+ * @internal
+ */
+export type getSourceFunction = (
+    htmlAttributes: Record<string, string>,
+    fragment?: DocumentFragment,
+    editor?: IEditor,
+    clipboardData?: ClipboardData
+) => KnownSourceType | false;
+
+const getSourceFunctions: getSourceFunction[] = [
+    isWordDesktopDocument,
+    isExcelDesktopDocument,
+    isPowerPointDesktopDocument,
+    documentContainWacElements,
+    isGoogleSheetDocument,
+    shouldConvertToSingleImage,
+];
+
+/**
+ * @internal
+ * This function tries to get the source of the Pasted content
+ * @param event the before paste event
+ * @param editor editor instance
+ * @returns The Type of pasted content, if no type found will return {KnownSourceType.Default}
+ */
+export default function getPasteSource(event: BeforePasteEvent, editor: IEditor): KnownSourceType {
+    const { htmlAttributes, clipboardData, fragment } = event;
+
+    for (const fn of getSourceFunctions) {
+        const result = fn(htmlAttributes, fragment, editor, clipboardData);
+        if (result) {
```

Comment:
This check is based on a fact that the valid value of KnownSourceType cannot be 0. It may be true now, but there is no build time check if some day we change it to have a valid 0 value, so it leave a hole to a potential bug.

Suggest we make the checker functions returns boolean, and store the function into a KnownSourceType=>function map, so we can easily return the KnownSourceType value when related function returns true.